src/alchemlyb/estimators/mbar_.py

```diff
@@ -187,9 +211,29 @@ def fit(self, u_nk):
         self.theta_ = pd.DataFrame(
             out["Theta"], columns=self._states_, index=self._states_
         )
+        if compute_entropy_enthalpy:
+            self._delta_h_ = pd.DataFrame(
+                out["Delta_u"], columns=self._states_, index=self._states_
+            )
+            self._d_delta_h_ = pd.DataFrame(
+                out["dDelta_u"], columns=self._states_, index=self._states_
+            )
+            self._delta_s_ = pd.DataFrame(
+                out["Delta_s"], columns=self._states_, index=self._states_
+            )
+            self._d_delta_s_ = pd.DataFrame(
+                out["dDelta_s"], columns=self._states_, index=self._states_
+            )
 
         self._delta_f_.attrs = u_nk.attrs
         self._d_delta_f_.attrs = u_nk.attrs
+        if compute_entropy_enthalpy:
+            self._delta_h_.attrs = u_nk.attrs
+            self._d_delta_h_.attrs = u_nk.attrs
+            self._delta_s_.attrs = u_nk.attrs
+            self._d_delta_s_.attrs = u_nk.attrs
+            self._delta_s_.attrs["energy_unit"] = "k"
```

Comment:
Why does entropy has unit of k? I think it shall be `u_nk.attrs["energy_unit"] + '/K'`?

Reply:
Personally, I'm fine for this unit to be the same as other energy unit as it would mean that the unit conversion would work as it is (between J and cal).

Reply:
Free energy is in units of energy (J) and when scaled by kT, is in units of kT. Similarly entropy is in units of energy/unit temperature (J/K) and so is scaled by k, with the units of k.

Reply:
Reduced entropy should be unitless, entropy should be J/degree, right?  Even if passed through some unitless quantities, probably the code should be clear what is going on.

Reply:
I think it's important to have the correct units, so I'll look into the unit conversion, I didn't consider that feature.

Reply:
I think we need to be a little careful when talking about things being in units of KT, this basically means unitless - all reduced quantities are unitless.  I think being in the units of KT, are "unitless here, I'm keeping track of if we were to convert them back to our physical system, what the units would be", though I'm rushing to get kids of to school so can't to into it carefully here with respect to the actual code.

Reply:
The way alchemlyb defines the units as `u_nk.attrs['energy_unit'] = "kT"` I think implies that it's dimensionless while also indicating what quantities should be used to restore the units (although I hope that an average user would know). This is also why I would want the entropy to similarly have the correct scaling quantity, 'k'.

Reply:
The `energy_unit`, allowed to be kT; kcal/mol and kJ/mol, it is just defaults to kT. I think `k` is more like a scientific constant for most people so having it as unit is kind of bad.
The unit conversion is supported between these three units, which is just a scaling factor.
So the easiest solution is to just set the entropy to have the same unit as enthalpy.

Reply:
To adjust the unit conversion functionality to accommodate entropy, should I add the appropriate duplicated function of `to_kT` as `to_k` or add if-statements to detect the case where it's entropy? I'm leaning toward the former since running a matrix of entropy values through `to_kT()` would feel wrong. This would mean I would duplicate `to_kcalmol(df, T=None)` as `to_kcalmolK(df)` and `to_kJmol(df, T=None)` as `to_kJmolK(df)`.